doc/user/content/sql/patterns/percentiles.md

```diff
@@ -0,0 +1,203 @@
+---
+title: "Tracking percentiles"
+description: "How to efficiently track percentiles in Materialize."
+menu:
+  main:
+    parent: 'sql-patterns'
+---
+
+Percentiles are often used to understand and interpret data. Informally speaking, the k-th percentile is the value such that k percent of all values are lower than the percentile.
+
+A naive way to compute percentiles is to simply order all values and pick the value at the position of the corresponding percentile. This can be easily expressed in Materialize through `order by`, `offset`, and `limit`. But this way of computing percentiles keeps all values around and therefore requires memory that linearly grows with the number of values that are tracked. There are better suited alternatives that reduce the required memory footprint, two of which are explained below: histograms and High Dynamic Range (HDR) histograms.
+
+Histograms have a reduced memory footprint that is linear in the number of *unique* values while they are still computing precise percentiles. HDR histograms further reduce the memory footprint at the expense of returning approximate percentiles only. They are particularly interesting if there is a long tail of large values that you want to track, which is often the case for latency meassurements.
+
+## Using histograms to compute percentiles
+
+Histograms reduce the required memory by tracking a count for each unique value instead of tracking all values. Given an `input` table or view, the historgram for `values` is defined as follows:
+
+```
+CREATE VIEW histogram AS
+SELECT
+  value AS bucket,
+  count(*) AS frequency
+FROM input
+GROUP BY value;
+```
+
+To query percentiles from that view, it's no longer possible to just order the `values` and pick a value from the right spot. Instead, the distribution of values needs to be reconstructed by determining the cumulative count (the sum of counts up through each bucket that came before) for each `bucket`. This is accomplished through a cross join the following view:
+
+```
+CREATE VIEW distribution AS
+SELECT
+  h.bucket,
+  h.frequency,
+  sum(g.frequency) AS cumulative_count,
+  sum(g.frequency) / (SELECT sum(frequency) FROM histogram) AS cumulative_distribution
+FROM histogram g, histogram h
+WHERE g.bucket <= h.bucket
+GROUP BY h.bucket, h.frequency
+ORDER BY cumulative_distribution;
+```
+
+The cumulative count and the cumulative distribution can then be used to query for arbitrary percentiles. The following query returns the 90-th percentile.
+
+```
+SELECT bucket AS percentile
+FROM distribution
+WHERE cumulative_distribution >= 0.9
+ORDER BY cumulative_distribution
+LIMIT 1;
+```
+
+To increase query performance, it can make sense to keep the `distribution` in memory by creating an index on the view:
```

Comment:
```suggestion
To increase query performance, it can make sense to keep the `distribution` always up to date in memory by creating an index on the view:
```